File: server/kotlin/src/main/kotlin/com/lebonpoint/shared/Configuration.kt
package com.lebonpoint.shared

/**
 * Application configuration
 * Loads configuration from environment variables with defaults
 */
object Configuration {
    /**
     * Server port
     */
    val port: Int by lazy {
        System.getenv("PORT")?.toIntOrNull() ?: 8080
    }

    /**
     * Database path
     * Relative to project root or absolute path
     */
    val databasePath: String by lazy {
        System.getenv("DATABASE_PATH") ?: "../database/db.sqlite"
    }

    /**
     * Application environment
     */
    val environment: Environment by lazy {
        when (System.getenv("ENV")?.lowercase()) {
            "production" -> Environment.PRODUCTION
            "testing" -> Environment.TESTING
            else -> Environment.DEVELOPMENT
        }
    }

    /**
     * Check if in development mode
     */
    val isDevelopment: Boolean
        get() = environment == Environment.DEVELOPMENT

    /**
     * Check if in testing mode
     */
    val isTesting: Boolean
        get() = environment == Environment.TESTING

    /**
     * Check if in production mode
     */
    val isProduction: Boolean
        get() = environment == Environment.PRODUCTION

    /**
     * Connection pool settings
     */
    val poolSize: Int by lazy {
        when (environment) {
            Environment.PRODUCTION -> 10
            Environment.TESTING -> 2
            Environment.DEVELOPMENT -> 5
        }
    }
}

/**
 * Application environment enum
 */
enum class Environment {
    DEVELOPMENT,
    TESTING,
    PRODUCTION
}

File: server/swift/Sources/App/Shared/Configuration.swift
import Foundation
import Vapor

/// Application configuration
struct Configuration: Sendable {
    let port: Int
    let databasePath: String
    let environment: Environment
    let logLevel: Logger.Level

    static let shared = Configuration()

    private init() {
        // Load from environment variables with defaults
        self.port = Int(Environment.get("PORT") ?? "9000") ?? 9000

        // Database path - relative to project or absolute
        let dbPathFromEnv = Environment.get("DATABASE_PATH")
        if let dbPath = dbPathFromEnv, !dbPath.isEmpty {
            // Check if it's a relative or absolute path
            if dbPath.hasPrefix("/") {
                self.databasePath = dbPath
            } else {
                // Resolve relative to the Swift server directory
                let currentDir = FileManager.default.currentDirectoryPath
                self.databasePath = "\(currentDir)/\(dbPath)"
            }
        } else {
            // Default to ../database/db.sqlite
            let currentDir = FileManager.default.currentDirectoryPath
            self.databasePath = "\(currentDir)/../database/db.sqlite"
        }

        // Environment detection
        let envString = Environment.get("ENVIRONMENT") ?? "development"
        switch envString.lowercased() {
        case "production":
            self.environment = .production
        case "testing":
            self.environment = .testing
        default:
            self.environment = .development
        }

        // Log level
        let logLevelString = Environment.get("LOG_LEVEL") ?? "info"
        switch logLevelString.lowercased() {
        case "debug":
            self.logLevel = .debug
        case "info":
            self.logLevel = .info
        case "warning":
            self.logLevel = .warning
        case "error":
            self.logLevel = .error
        case "critical":
            self.logLevel = .critical
        default:
            self.logLevel = .info
        }
    }
}
